tools/codegen/api/unboxing.py

```diff
@@ -0,0 +1,201 @@
+from typing import List, Tuple
+
+from tools.codegen.api import cpp
+from tools.codegen.api.types import Binding, CType, CppSignatureGroup
+from tools.codegen.model import (
+    Argument,
+    NativeFunction,
+    Type,
+    BaseType,
+    OptionalType,
+    ListType,
+    BaseTy,
+)
+
+# This file generates the code for unboxing wrappers, i.e., the glue logic to unbox a boxed operator and convert the
+# ivalues from stack to correct arguments to the unboxed kernel, based on corresponding JIT schema. This codegen is
+# an alternative way to generate unboxing wrappers similar to the existing C++ metaprogramming approach but gets the
+# job done statically. These generated unboxing wrappers will be useful under the scenario where we need to register
```

Comment:
* How are we validating that the metaprogrammed templates are equivalent to the code generated unboxing wrappers here?
* If I change some logic here, how does it have to be reflected in the C++ metaprogram? (And vice versa?)

Reply:
> How are we validating that the metaprogrammed templates are equivalent to the code generated unboxing wrappers here?

I'm adding tests to cover the types for mobile use cases.

> If I change some logic here, how does it have to be reflected in the C++ metaprogram? (And vice versa?)

These two are not coupled right now. We didn't think of the need to couple them together, since this is used by *mobile only* and completely separated from server runtime.

Reply:
> These two are not coupled right now. We didn't think of the need to couple them together, since this is used by mobile only and completely separated from server runtime.

This is an interesting claim. While I certainly agree that the modification here affords some amount of decoupling (e.g., if server adds a problematic feature that mobile doesn't need, codegen unboxing can choose not to support that functionality for now), it seems that to keep our system coherent, we want the C++ metaprogram and the codegen to "do the same thing" and that means keeping both sides of the system synchronized if you make a change. This isn't hypothetical; for example https://github.com/pytorch/pytorch/pull/67964 is the kind of change we're interested in making to the calling convention.

Reply:
I see. I am planning to add the same set of tests to metaprogramming and codegen unboxing, hopefully these tests can serve the purpose of making sure they are synchronized.